packages/labs/ssr-dom-shim/src/element-internals.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * TODO
+ * - This type could be better inferred as Record<keyof ARIAMixin, string>;
+ *   however, modern browsers and TypeScript seem to lack a common
+ *   definition of the keys listed in ARIAMixin
+ */
+export const ariaMixinEnum: Record<string, string> = {
+  ariaAtomic: 'aria-atomic',
+  ariaAutoComplete: 'aria-autocomplete',
+  ariaBraileLabel: 'aria-brailelabel',
+  ariaBraileDescription: 'aria-brailedescription',
+  ariaBusy: 'aria-busy',
+  ariaChecked: 'aria-checked',
+  ariaColCount: 'aria-colcount',
+  ariaColIndex: 'aria-colindex',
+  ariaColSpan: 'aria-colspan',
+  ariaCurrent: 'aria-current',
+  ariaDescription: 'aria-description',
+  ariaDisabled: 'aria-disabled',
+  ariaExpanded: 'aria-expanded',
+  ariaHasPopup: 'aria-haspopup',
+  ariaHidden: 'aria-hidden',
+  ariaInvalid: 'aria-invalid',
+  ariaKeyShortcuts: 'aria-keyshortcuts',
+  ariaLabel: 'aria-label',
+  ariaLevel: 'aria-level',
+  ariaLive: 'aria-live',
+  ariaModal: 'aria-modal',
+  ariaMultiLine: 'aria-multiline',
+  ariaMultiSelectable: 'aria-multiselectable',
+  ariaOrientation: 'aria-orientation',
+  ariaPlaceholder: 'aria-placeholder',
+  ariaPosInSet: 'aria-posinset',
+  ariaPressed: 'aria-pressed',
+  ariaReadOnly: 'aria-readonly',
+  ariaRequired: 'aria-required',
+  ariaRoleDescription: 'aria-roledescription',
+  ariaRowCount: 'aria-rowcount',
+  ariaRowIndex: 'aria-rowindex',
+  ariaRowSpan: 'aria-rowspan',
+  ariaSelected: 'aria-selected',
+  ariaSetSize: 'aria-setsize',
+  ariaSort: 'aria-sort',
+  ariaValueMax: 'aria-valuemax',
+  ariaValueMin: 'aria-valuemin',
+  ariaValueNow: 'aria-valuenow',
+  ariaValueText: 'aria-valuetext',
+  role: 'role',
+};
+
+// Shim the global element internals object
+// Methods should be fine as noops and properties can generally
+// be while on the server.
+export const InternalsShim = class ElementInternals implements ARIAMixin {
```

Comment:
I'm getting
```
Class 'ElementInternals' incorrectly implements interface 'ARIAMixin'.
  Type 'ElementInternals' is missing the following properties from type 'ARIAMixin': ariaColIndexText, ariaRowIndexText ts(2420)
```
Is this expected?

Reply:
Also, maybe related, is it possible to dedupe the class fields listing below with the enum above?